src/main/java/bookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -49,10 +49,19 @@ private String getErrorMessage(ObjectError e) {
 
     @ExceptionHandler({EntityNotFoundException.class})
     protected ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException ex) {
+        return getResponseEntity(ex.getMessage(), ex);
+    }
+
+    @ExceptionHandler({RegistrationException.class})
+    protected ResponseEntity<Object> handleRegistrationException(RegistrationException ex) {
+        return getResponseEntity(ex.getMessage(), ex);
+    }
+
+    private ResponseEntity<Object> getResponseEntity(String message, Exception ex) {
         Map<String, Object> body = new LinkedHashMap<>();
         body.put(MAP_TIME_KEY, LocalDateTime.now());
         body.put(MAP_STATUS_KEY, HttpStatus.NOT_FOUND);
-        body.put(MAP_ERROR_KEY, ex.getMessage());
+        body.put(MAP_ERROR_KEY, message);
```

Comment:
Why do you return an exception but not the message exception as it was before?
Are there any changes in response?

Reply:
because i delegated creation of this method to Ide)